### PR TITLE
Add support for extra security groups annotation (NLB)

### PR DESCRIPTION
**Pull request by omerap12**
### Issue
#3679

### Description
Add the annotation `service.beta.kubernetes.io/aws-load-balancer-extra-security-groups` to attach an existing security group to the NLB.

Add a loop in pkg/service/model_build_load_balancer.go to retrieve all security groups from the annotation `service.beta.kubernetes.io/aws-load-balancer-extra-security-groups` and append them to the existing security groups list.

Made local testing, for the following yaml fille:
```bash
apiVersion: v1
kind: Service
metadata:
  name: nlb-sample-service
  namespace: default
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-name: "test-new-annotation"
    service.beta.kubernetes.io/aws-load-balancer-extra-security-groups: sg-0b036fd0124715a4c
    service.beta.kubernetes.io/aws-load-balancer-security-groups: sg-0c721856c63fba26b
spec:
  ports:
    - port: 80
      targetPort: 80
      protocol: TCP
  type: LoadBalancer
  selector:
    app: nginx
```
And this:
```bash
apiVersion: v1
kind: Service
metadata:
  name: nlb-sample-service
  namespace: default
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-name: "test-new-annotation"
    service.beta.kubernetes.io/aws-load-balancer-extra-security-groups: test-for-alb-2
    service.beta.kubernetes.io/aws-load-balancer-security-groups: sg-0c721856c63fba26b
spec:
  ports:
    - port: 80
      targetPort: 80
      protocol: TCP
  type: LoadBalancer
  selector:
    app: nginx
```

The following nlb was created:
image

As you can see, the extra sg was added to the security groups list.
### Checklist
- [ ] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `docs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: docs/guide/service/annotations.md [hunk 1]
```diff
@@ -49,7 +49,8 @@
...
 | [service.beta.kubernetes.io/aws-load-balancer-target-node-labels](#target-node-labels)           | stringMap               |                           |                                                        |
 | [service.beta.kubernetes.io/aws-load-balancer-attributes](#load-balancer-attributes)             | stringMap               |                           |                                                        |
-| [service.beta.kubernetes.io/aws-load-balancer-security-groups](#security-groups)                 | stringList              |                           |                                                        | 
+| [service.beta.kubernetes.io/aws-load-balancer-security-groups](#security-groups)                 | stringList              |                           |                                                        |
+| [service.beta.kubernetes.io/aws-load-balancer-extra-security-groups](##security-groups)          | stringList              |                           |                                                        |
```
Comment (andreybutenko) [inline docs/guide/service/annotations.md:53]: Should this have a unique anchor tag? Like `#extra-security-groups`

**Review by andreybutenko — CHANGES_REQUESTED** [review]
Thanks for the contribution! Left one small comment, sharing your PR internally for additional reviewers

File: docs/guide/service/annotations.md (hunk 1 above)
Comment (omerap12) [reply docs/guide/service/annotations.md:53]: got it, adding

**Comment by omerap12** [comment]
I don't think the failures are due to my PR, right?

**Comment by omerap12** [comment]
> Thanks for the contribution! Left one small comment, sharing your PR internally for additional reviewers

Hey, any update?

**Review by andreybutenko — COMMENTED** [review]
Thanks for the contribution! It looks good to me, but I'm pinging some maintainers to review it too :)

**Comment by andreybutenko** [comment]
Hi @oliviassss @wweiwei-li , could you please review this PR too? It is related to #3679 which we previously discussed. Thank you! :)

**Review by andreybutenko — APPROVED** [review]

**Comment by omerap12** [comment]
Friendly reminder :) @oliviassss

**Comment by oliviassss** [comment]
@omerap12, sorry for the delayed reply. I'm a bit confused on the use-case.
so if you're already using `service.beta.kubernetes.io/aws-load-balancer-security-groups`, which means the controller is not managing the SGs for you. why not just attach extra SGs in this annotation, like
```
service.beta.kubernetes.io/aws-load-balancer-security-groups: sg0, extraSG0, extraSG1
```
I'm just trying to avoid redundant annotations here. Thanks

**Pull request closed without merging**